### ISSUE TITLE
[telemetry] Instrument APIInvoker with spans and metrics

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#39]
## Goal

Instrument REST API invocation in `ph-oton-api` with a ph-telemetry `SERVER` span plus counters and a duration histogram.

This is the single highest-value instrumentation point in ph-oton: it is where every REST call the application serves passes through.

## Where

`ph-oton-api/src/main/java/com/helger/photon/api/APIInvoker.java`

```java
public void invoke (@NonNull final InvokableAPIDescriptor aInvokableDescriptor,
                    @NonNull final IRequestWebScopeWithoutResponse aRequestScope,
                    @NonNull final UnifiedResponse aUnifiedResponse) throws Exception
```

The method already declares `throws Exception`, so `Telemetry.withSpanVoidThrowing (..., Exception)` fits without any signature change.

The method already has the exact shape needed: a `StopWatch` started at the top, a `try` / `catch (Exception)` / `finally`, and three `StatisticsManager` handlers. **Keep the statistics handlers as they are** and add telemetry alongside them.

## The cardinality trap — read this before writing code

`InvokableAPIDescriptor.getPath ()` is documented as *"The URL path requested by the user, relative to the servlet"* — it is the **concrete** path with all path variables already substituted (`/user/4711`). It is unbounded and must **never** become a metric attribute. The existing `STATS_FUNCTION_INVOKE.increment (sPath)` already has this problem; do not copy it.

The low-cardinality dimension is the **path template**:

```java
aInvokableDescriptor.getAPIDescriptor ().getPathDescriptor ().getAsURLString ()   // e.g. "/user/{id}"
```

- Path **template** → metric attribute *and* span attribute.
- Concrete path → span attribute only.

`PathDescriptor.getAsURLString ()` exists in `com.helger.photon.api.pathdescriptor.PathDescriptor`.

## What to emit

New `com.helger.photon.api.CAPITelemetry`:

| Constant | Value |
|---|---|
| `SPAN_INVOKE` | `photon.api.invoke` |
| `METRIC_INVOCATIONS` | `photon.api.invocations` |
| `METRIC_DURATION` | `photon.api.duration` |
| `ATTR_API_ROUTE` | `photon.api.route` (path **template** — low cardinality) |
| `ATTR_API_PATH` | `photon.api.path` (concrete path — **span only**) |
| `ATTR_API_METHOD` | `photon.api.method` (HTTP method) |
| `ATTR_API_SUCCESS` | `photon.api.success` |
| `ATTR_API_EXCEPTION_HANDLED` | `photon.api.exception.handled` |
| `UNIT_REQUEST` | `{request}` |
| `UNIT_MILLIS` | `ms` |

New `com.helger.photon.api.APIMetrics` holding `public static final` instruments (counter `METRIC_INVOCATIONS` unit `{request}`, histogram `METRIC_DURATION` unit `ms`), created once at class load, mirroring `LongRunningJobMetrics`.

New package-private `com.helger.photon.api.APITelemetry` emitter, mirroring `LongRunningJobTelemetry`.

**Span** `photon.api.invoke`, kind `SERVER`, wrapping the whole body of `invoke (...)`:
- attributes: route, concrete path, HTTP method
- exception recorded automatically by `withSpanVoidThrowing`; set the status explicitly to `ERROR` when the exception mapper handled the exception (the method returns normally in that case, so the helper would otherwise mark it OK)

**Counter** `photon.api.invocations` incremented once per invocation with attributes `route`, `method`, `success` — one increment on *every* path through the method, success and failure alike, so the counter is a true invocation total.

**Histogram** `photon.api.duration` (ms) recorded in the existing `finally` block from `aSW.stopAndGetMillis ()`, with attributes `route`, `method`, `success`.

The HTTP method comes from the descriptor, not the request scope:

```java
aInvokableDescriptor.getAPIDescriptor ().getHttpMethod ().getName ()   // com.helger.http.EHttpMethod implements IHasName
```

## Exception handling detail

The `catch (Exception ex)` branch runs the `IAPIExceptionMapper`. When the mapper reports the exception as handled, `invoke (...)` returns normally — so the span must still be marked as failed and the counter must record `success=false`, with `photon.api.exception.handled=true` on the span to distinguish a mapped error response from a propagated one.

## POM

`ph-oton-api/pom.xml` needs a new dependency (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Span, counter and histogram emitted as specified; no unbounded value on any metric attribute.
- [ ] Existing `StatisticsManager` handlers untouched.
- [ ] A unit test in the style of `ph-oton-mgrs/src/test/java/.../longrun/LongRunningJobTelemetryTest.java` — install recording `ITelemetryTracerSPI` / `ITelemetryMeterSPI` via `Telemetry.install (...)` / `TelemetryMetrics.install (...)`, reset both to `null` in `@After`, and assert on span name/attributes and instrument values for a successful call, a mapper-handled exception and a propagated exception.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-api/.../api/CAPITelemetry.java` | new, public - span / metric / attribute names |
| `ph-oton-api/.../api/APIMetrics.java` | new, public - counter `photon.api.invocations`, histogram `photon.api.duration` |
| `ph-oton-api/.../api/APITelemetry.java` | new, package-private emitter |
| `ph-oton-api/.../api/APIInvoker.java` | modified |
| `ph-oton-api/pom.xml` | `com.helger.telemetry:ph-telemetry` added (no `<version>`, BOM) |
| `ph-oton-api/src/test/.../APITelemetryTest.java` | new, 3 tests |

The naming follows the reference implementation: `<Feature>Metrics` / `<Feature>Telemetry` next to the instrumented class, with only the module constants class hoisted to the module root package as decided.

### Emission

`invoke (...)` is wrapped in `Telemetry.<Exception> withSpanVoidThrowing (CAPITelemetry.SPAN_INVOKE, ETelemetrySpanKind.SERVER, ...)`. No signature change was needed - the method already declared `throws Exception`.

* Span `photon.api.invoke` (`SERVER`) with `photon.api.route`, `photon.api.path`, `photon.api.method`, `photon.api.success` and - on a mapper-handled exception - `photon.api.exception.handled`.
* Counter `photon.api.invocations` and histogram `photon.api.duration` are both emitted from the **existing `finally` block**, so every code path - success, mapper-handled failure, propagated failure and even an `Error` - is counted exactly once.

### Cardinality

The trap described in the issue is handled and asserted in the test:

* `photon.api.route` = `aInvokableDescriptor.getAPIDescriptor ().getPathDescriptor ().getAsURLString ()` → `/user/{id}`. Metric **and** span attribute.
* `photon.api.path` = `aInvokableDescriptor.getPath ()` → `/user/4711`. **Span attribute only.** `APITelemetryTest.testSuccessfulInvocation` explicitly asserts that the metric attribute map has exactly 3 entries and contains no `photon.api.path`.
* `photon.api.method` = `aInvokableDescriptor.getAPIDescriptor ().getHttpMethod ().getName ()`.

The pre-existing `STATS_FUNCTION_INVOKE.increment (sPath)` keyed by the concrete path was **left untouched** as decided in #55.

### Exception mapper

When the `IAPIExceptionMapper` reports the exception as handled, `invoke (...)` returns normally. `APITelemetry.onInvokeError (aSpan, ex, true)` therefore records the exception on the span and sets the status to `ERROR` itself, because `withSpanVoidThrowing` never sees the exception in that case. The counter records `photon.api.success=false`. Covered by `testMappedException`.

### Verification

Full reactor `mvn install` on all 23 modules: BUILD SUCCESS, no test failures. `APITelemetryTest`: 3/3 green. `README.md` news bullet added under the new `v10.5.1 - work in progress` section (10.5.0 was released in the meantime).
